### PR TITLE
fix potential namespace conflict

### PR DESCRIPTION
**Pull request by rafammpp**
From this issue: https://github.com/taylorhakes/python-redis-cache/issues/14

### CODE REVIEW

**Comment by rafammpp** [comment]
I ran the test and got an assertion error because of the new key format. I fixed it using the new expected key, you can review it in the last commit.
Now I am going to write some tests for methods and inner functions.

File: tests/test_redis_cache.py [hunk 1]
```diff
@@ -238,3 +238,47 @@ def add_basic_get(arg1, arg2):
...
+        @staticmethod
+        @cache.cache()
+        def static_method():
+            return 'B'
+    
+    A.static_method()
+    B.static_method()
+
+    # If there was a key conflict, the results should be the same
+    assert A.static_method() != B.static_method()
```
Comment (taylorhakes) [inline tests/test_redis_cache.py:260]: Let's also verify that 2 keys exist in redis

File: tests/test_redis_cache.py [hunk 2]
```diff
@@ -238,3 +238,47 @@ def add_basic_get(arg1, arg2):
...
+            return 'B'
+
+        inner_function()
+        return inner_function
+    
+    first_func = a()
+    second_func = b()
+
+    # If there was a key conflict, the results should be the same
+    assert first_func() != second_func()
```
Comment (taylorhakes) [inline tests/test_redis_cache.py:284]: Can you run these methods twice instead of just once? The first time won't cache. Let's also verify the keys.

**Comment by rafammpp** [comment]
Done!
For readability, I've written the first call of the inner functions outside.

I make use of the get_key function instead of hard coded keys. Is that ok?

**Comment by taylorhakes** [comment]
This looks great! We just need to be careful since this is a breaking change. I am thinking we need to create a function called `copy_old_data()` and `delete_old_data()` or something similar, which copys the cached data from the old key name to the new. Otherwise some people may see a ton of cache misses when they upgrade because the key names have suddenly changed.

**Comment by taylorhakes** [comment]
@rafammpp I am not suggesting we upgrade everyone automatically. We should just add a utility for people to do it manually (the functions I mentioned above).

Yes, someone could have a data conflict, but they already have it today. It will not be adding to the issue at all. People can obviously decide not to move keys and just let the cache miss as you are doing. 
Having everyone's keys be invalid randomly on upgrade with no fix is a serious issue. In your use case it might be Ok to have things miss cache, but for a lot of people they will never upgrade because that is a deal breaker. 

I cannot merge this PR without a smooth upgrade path for people that can't have lots of things miss cache.

**Comment by rafammpp** [comment]
I've done it. I still think is too risky to migrate data, there are way to much cases. But it is done.

**Comment by rafammpp** [comment]
@taylorhakes How is it going? (No presure, just checking)

**Pull request merged**